### PR TITLE
Revert OGRE-Next custom support and back to use IgnOGRE2 module

### PR DESCRIPTION
**Pull request by j-rivero**
# 🎉 New feature

The PR mostly removes the support introduced in #577 to workaround the problems with Ubuntu Jammy using ogre-next package. The support is implemented in https://github.com/ignitionrobotics/ign-cmake/pull/224 that should work transparently for ign-rendering.

The PR tries to look for OGRE2 twice, one with PlanarRelections component (needed for Jammy) and the other one without PlanarReflections. The issues is being worked in https://github.com/ignitionrobotics/ign-rendering/issues/597
 
## Test it

Colcon workspace with https://github.com/ignitionrobotics/ign-cmake/pull/224 and this PR should be enough.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by j-rivero** [comment]
I think that we are going to need a new release of ign-cmake to be able to check CI.

File: CMakeLists.txt [hunk 1]
```diff
@@ -96,11 +96,21 @@ if (OGRE_FOUND)
...
-# Find OGRE2
+# Find OGRE2: first try to find OGRE2 built with PlanarReflections support and
+# fallback to look for OGRE2 without it. Both seems to works for ign-rendering.
+# See https://github.com/ignitionrobotics/ign-rendering/issues/597
 ign_find_package(IgnOGRE2 VERSION 2.2.0
+    COMPONENTS HlmsPbs HlmsUnlit Overlay PlanarReflections
+    PRIVATE_FOR ogre2
+    QUIET)
+
+if (NOT OGRE2_FOUND)
```
Comment (iche033) [inline CMakeLists.txt:107]: maybe check to see if `${OGRE2-PlanarReflections}` equals `"OGRE2-PlanarReflections-NOTFOUND"` here?

Comment (j-rivero) [reply CMakeLists.txt]: uh much better, thanks Ian [3fd7e1d](https://github.com/ignitionrobotics/ign-rendering/pull/605/commits/3fd7e1d06f79454000e9ca0021c5b9f54750abe7).

**Comment by j-rivero** [comment]
I think that CI is going to require new ignition cmake to work. We need to bump the ign-cmake version required here too.

**Comment by j-rivero** [comment]
Ummm, we have a problem that is hard to solve or workaround in ign-rendering. The two calls for `ign_find_package` is completely buggy. We would need https://github.com/ignitionrobotics/ign-cmake/pull/231.

**Review by chapulina — COMMENTED** [review]
Nice! I see this on the Bionic and Focal builds:

```
  -- Looking for OGRE using the name: OGRE2
  --   + component HlmsPbs: found
  --   + component HlmsUnlit: found
  --   + component Overlay: found
  --   ! component PlanarReflections: not found!
  --   ! OGRE2 not found
  -- Looking for OGRE using the name: OGRE-Next
  --   ! OGRE-Next not found
  -- Looking for IgnOGRE2 - not found
  
  -- PlanarReflections component was not found. Try looking without it:
  -- Looking for OGRE using the name: OGRE2
  --   + component HlmsPbs: found
  --   + component HlmsUnlit: found
  --   + component Overlay: found
  -- Looking for IgnOGRE2 - found
```

It looks like Jammy CI is not on `ign-rendering6` anymore and was reverted with a commit directly to the branch:

https://github.com/ignitionrobotics/ign-rendering/commit/92bbff379ab3a7915f59373bd26b6804de77575e

We should make sure this patch works with Jammy. Can we add back that CI here?

**Comment by j-rivero** [comment]
> We should make sure this patch works with Jammy. Can we add back that CI here?

I made a branch / commit on top of this branch https://github.com/ignitionrobotics/ign-rendering/commit/619c9b6a85bdafb86138a40b1edadca33b2d6e0a.  CI is happy. I'll follow up a PR with it, want to work in parallel releasing this change.

> It looks like Jammy CI is not on `ign-rendering6` anymore and was reverted with a commit directly to the branch:

my bad sorry, wrong branch.

**Review by chapulina — APPROVED** [review]
Nice! I see Jammy finds it now:

```
   -- Looking for OGRE using the name: OGRE2
  --   ! OGRE2 not found
  -- Looking for OGRE using the name: OGRE-Next
  --   + component HlmsPbs: found
  --   + component HlmsUnlit: found
  --   + component Overlay: found
  --   + component PlanarReflections: found
  -- Looking for IgnOGRE2 - found
```

🚢 it!

**Pull request merged**